File: frontend/src/types/ticket.ts
export interface Ticket {
  id: string
  external_id: string | null
  subject: string
  body: string
  client_name: string | null
  client_segment: string | null
  source_channel: string | null
  status: string
  raw_address: string | null
  created_at: string
  updated_at: string
}

export interface TicketAI {
  id: string
  ticket_id: string
  type: string | null
  sentiment: string | null
  priority_1_10: number | null
  lang: string
  summary: string | null
  recommended_actions: string[]
  lat: number | null
  lon: number | null
  geo_status: string
  confidence_type: number | null
  confidence_sentiment: number | null
  confidence_priority: number | null
  enriched_at: string | null
}

export interface TicketAssignment {
  id: string
  ticket_id: string
  manager_id: string
  business_unit_id: string
  assigned_at: string
  routing_reason: string | null
  is_current: boolean
}

export interface AuditLog {
  id: string
  ticket_id: string
  step: string
  input_data: unknown
  output_data: unknown
  decision: string
  candidates: string[] | null
  created_at: string
}

export interface TicketWithDetails {
  ticket: Ticket
  ai: TicketAI | null
  assignment: TicketAssignment | null
  assigned_manager: Manager | null
  audit_trail: AuditLog[]
}

export interface Manager {
  id: string
  full_name: string
  email: string | null
  business_unit_id: string
  is_vip_skill: boolean
  is_chief_spec: boolean
  languages: string[]
  max_load: number
  current_load: number
  is_active: boolean
  created_at: string
}

export interface ManagerWithOffice extends Manager {
  office_name: string
  office_city: string
  utilization_pct: number
}

File: backend/internal/domain/ticket.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Ticket struct {
	ID            uuid.UUID  `json:"id" db:"id"`
	ExternalID    *string    `json:"external_id" db:"external_id"`
	Subject       string     `json:"subject" db:"subject"`
	Body          string     `json:"body" db:"body"`
	ClientName    *string    `json:"client_name" db:"client_name"`
	ClientSegment *string    `json:"client_segment" db:"client_segment"`
	SourceChannel *string    `json:"source_channel" db:"source_channel"`
	Status        string     `json:"status" db:"status"`
	RawAddress    *string    `json:"raw_address" db:"raw_address"`
	ManagerID     *uuid.UUID `json:"manager_id,omitempty" db:"manager_id"`
	OfficeID      *uuid.UUID `json:"office_id,omitempty" db:"office_id"`
	CreatedAt     time.Time  `json:"created_at" db:"created_at"`
	UpdatedAt     time.Time  `json:"updated_at" db:"updated_at"`
}

type TicketAI struct {
	ID                  uuid.UUID  `json:"id" db:"id"`
	TicketID            uuid.UUID  `json:"ticket_id" db:"ticket_id"`
	Type                *string    `json:"type" db:"type"`
	Sentiment           *string    `json:"sentiment" db:"sentiment"`
	Priority110         *int       `json:"priority_1_10" db:"priority_1_10"`
	Lang                string     `json:"lang" db:"lang"`
	Summary             *string    `json:"summary" db:"summary"`
	RecommendedActions  []byte     `json:"recommended_actions" db:"recommended_actions"`
	Lat                 *float64   `json:"lat" db:"lat"`
	Lon                 *float64   `json:"lon" db:"lon"`
	GeoStatus           string     `json:"geo_status" db:"geo_status"`
	ConfidenceType      *float64   `json:"confidence_type" db:"confidence_type"`
	ConfidenceSentiment *float64   `json:"confidence_sentiment" db:"confidence_sentiment"`
	ConfidencePriority  *float64   `json:"confidence_priority" db:"confidence_priority"`
	EnrichedAt          *time.Time `json:"enriched_at" db:"enriched_at"`
	CreatedAt           time.Time  `json:"created_at" db:"created_at"`
}

type TicketListFilter struct {
	Page      int
	PerPage   int
	Status    string
	Sentiment string
	Segment   string
	Type      string
	Lang      string
	Search    string
}

type TicketWithDetails struct {
	Ticket     Ticket            `json:"ticket"`
	AI         *TicketAI         `json:"ai"`
	Assignment *TicketAssignment `json:"assignment"`
	Manager    *ManagerWithOffice `json:"assigned_manager"`
	AuditTrail []AuditLog        `json:"audit_trail"`
}

// EnrichmentResult is the payload from n8n callback.
type EnrichmentResult struct {
	TicketID            uuid.UUID `json:"ticket_id"`
	Type                string    `json:"type"`
	Sentiment           string    `json:"sentiment"`
	Priority110         int       `json:"priority_1_10"`
	Lang                string    `json:"lang"`
	Summary             string    `json:"summary"`
	RecommendedActions  []string  `json:"recommended_actions"`
	Lat                 *float64  `json:"lat"`
	Lon                 *float64  `json:"lon"`
	GeoStatus           string    `json:"geo_status"`
	ConfidenceType      float64   `json:"confidence_type"`
	ConfidenceSentiment float64   `json:"confidence_sentiment"`
	ConfidencePriority  float64   `json:"confidence_priority"`
}
